### PR TITLE
mbedtls: Update to upstream version 2.28.4

### PR DESCRIPTION
**Pull request by akien-mga**
Release notes: https://github.com/Mbed-TLS/mbedtls/releases/tag/v2.28.4

Bug fixes and minor enhancements. No security advisories.

### CODE REVIEW

**Review by YuriSizov — APPROVED** [review]

File: thirdparty/README.md [hunk 1]
```diff
@@ -370,7 +370,7 @@ Patch `godot-node-debug-fix.patch` workarounds shadowing of godot's Node class i
 ## mbedtls
 
 - Upstream: https://github.com/Mbed-TLS/mbedtls
-- Version: 2.18.3 (981743de6fcdbe672e482b6fd724d31d0a0d2476, 2023)
+- Version: 2.18.4 (aeb97a18913a86f051afab11b2c92c6be0c2eb83, 2023)
```
Comment (YuriSizov) [inline thirdparty/README.md:373]: Isn't it 2.28.4?

Comment (akien-mga) [reply thirdparty/README.md]: Fixed.

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.3.

**Comment by YuriSizov** [comment]
Cherry-picked for 4.1.2.

**Pull request merged**